### PR TITLE
fix(discover2) Fix next and prev Link headers

### PR DESCRIPTION
**Pull request by evanh**
The base API only appends one of the query params for each query key. Add a new
version that adds all the values to ensure that all the fields get passed in.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -91,6 +94,29 @@ def handle_results(self, request, organization, project_ids, results):
...
 class OrganizationEventsV2Endpoint(OrganizationEventsEndpointBase):
+    def build_cursor_link(self, request, name, cursor):
+        # The base API function only uses the last query parameter, but this endpoint
+        # needs all the parameters, particularly for the "field" query param.
+        querystring = u"&".join(
+            u"{0}={1}".format(urlquote(query[0]), urlquote(value))
+            for query in request.GET.lists()
+            if query[0] != "cursor"
+            for value in query[1]
+        )
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: You should be able to use `six.moves.urllib.parse.urlencode(params, doseq=True)` to encode `request.GET` if you copy the dictionary and remove cursor first.

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 2]
```diff
@@ -803,6 +804,49 @@ def test_condition_on_aggregate_misses(self):
...
+        print (response["Link"])
+        links = parse_link_header(response["Link"])
+        for link in links:
+            assert "field=issue.id" in link
+            assert "field=count%28id%29" in link
+            assert "field=context.key" in link
+            assert "sort=-count_id" in link
+            assert "query=language%3AC%2B%2B" in link
+
+        print (links)
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events_v2.py]: Linter will complain about this.

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 3]
```diff
@@ -803,6 +804,49 @@ def test_condition_on_aggregate_misses(self):
...
+                format="json",
+                data={
+                    "field": ["count(id)", "issue.id", "context.key"],
+                    "sort": "-count_id",
+                    "query": "language:C++",
+                },
+            )
+
+        assert response.status_code == 200, response.content
+        print (response["Link"])
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events_v2.py]: Linter will complain about this.

File: src/sentry/api/endpoints/organization_events.py (hunk 1 above)
Comment (evanh) [reply src/sentry/api/endpoints/organization_events.py]: Is that better?

Comment (markstory) [reply src/sentry/api/endpoints/organization_events.py]: It would use fewer lines of code, but this works too :smile:

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**